### PR TITLE
doc: improvements on the docstring of ModelStep

### PR DESCRIPTION
**Pull request by navaj0**
*Issue #, if available:* 3129

*Description of changes:*
 * remove the README under the workflow folder, which is outdated. 
 * add back the docstring of RegisterModel step
 * polish the docstring of ModelStep

*Testing done:*

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [x ] I have read the [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md) doc
- [x] I certify that the changes I am introducing will be backword compatible, and I have discussed concerns about this, if any, with the Python SDK team
- [x] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md#committing-your-change)
- [x] I have passed the region in to all S3 and STS clients that I've initialized as part of this change.
- [x] I have updated any necessary documentation, including [READMEs](https://github.com/aws/sagemaker-python-sdk/blob/master/README.rst) and [API docs](https://github.com/aws/sagemaker-python-sdk/tree/master/doc) (if appropriate)

#### Tests

- [x] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [x] I have added unit and/or integration tests as appropriate to ensure backward compatibility of the changes
- [x] I have checked that my tests are not configured for a specific region or account (if appropriate)
- [x] I have used [`unique_name_from_base`](https://github.com/aws/sagemaker-python-sdk/blob/master/src/sagemaker/utils.py#L77) to create resource names in integ tests (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by jerrypeng7773 — APPROVED** [review]

File: src/sagemaker/model.py [hunk 1]
```diff
@@ -399,15 +401,22 @@ def create(
...
-                https://boto3.amazonaws.com/v1/documentation
-                /api/latest/reference/services/sagemaker.html#SageMaker.Client.add_tags
+                the model (default: None). Example::
+
+                    tags = [{'Key': 'tagname', 'Value':'tagvalue'}]
+
+                For more information about tags, see
+                `boto3 documentation <https://boto3.amazonaws.com/v1/documentation/\
+api/latest/reference/services/sagemaker.html#SageMaker.Client.add_tags>`_
+
```
Comment (qidewenwhen) [inline src/sagemaker/model.py:416]: We have checked this in the pages built by sphinx right?

File: src/sagemaker/workflow/model_step.py [hunk 2]
```diff
@@ -50,10 +50,17 @@ def __init__(
...
-            depends_on (List[Union[str, Step, StepCollection]]): A list of `Step`/`StepCollection`
-                names or `Step` instances or `StepCollection` instances that the first step,
-                in this `ModelStep` collection, depends on.
+            step_args (_ModelStepArguments): The arguments for the `ModelStep` definition,
+                generated by invoking the :func:`~sagemaker.model.Model.register` or
+                :func:`~sagemaker.model.Model.create`
+                under the :class:`~sagemaker.workflow.pipeline_context.PipelineSession`. Example::
+
+                     model = Model(session=PipelineSession())
+                     model_step = ModelStep(step_args=model.register())
```
Comment (qidewenwhen) [inline src/sagemaker/workflow/model_step.py:59]: model = Model(sagemaker_session=PipelineSession()) ?

File: src/sagemaker/model.py (hunk 1 above)
Comment (navaj0) [reply src/sagemaker/model.py:416]: yes

File: src/sagemaker/workflow/model_step.py (hunk 2 above)
Comment (navaj0) [reply src/sagemaker/workflow/model_step.py:59]: updated it.

**Review by qidewenwhen — APPROVED** [review]

**Review by BasilBeirouti — APPROVED** [review]

**Review by mufaddal-rohawala — APPROVED** [review]

**Pull request merged**